File: Exercises/rb101-rb109_small_problems/easy_8/double_char_2.rb
def double_consonants(string)
  new_str = ''
  string.each_char do |char|
    (/[^aeiouAEIOU\s\d\W]/).match?(char) ? new_str << char * 2 : new_str << char
  end
  new_str
end

puts double_consonants('String') == "SSttrrinngg"
puts double_consonants("Hello-World!") == "HHellllo-WWorrlldd!"
puts double_consonants("July 4th") == "JJullyy 4tthh"
puts double_consonants('') == ""

File: js210/exercises/easy_5/double_char_2.js
function doubleConsonants(string) {
  let consonantRegExp = /[^aeiou]/gi;
  return string.replace(consonantRegExp, match => match + match);
}

console.log(doubleConsonants('String'));          // "SSttrrinngg"
console.log(doubleConsonants('Hello-World!'));    // "HHellllo-WWorrlldd!"
console.log(doubleConsonants('July 4th'));        // "JJullyy 4tthh"
console.log(doubleConsonants(''));                // ""
